File: C++/Easy/Accepted/rings-and-rods_8.cpp
class Solution {
public:
    int countPoints(string rings) {
        
        int n = rings.size();
        int m = n/2;
        int ans = 0;
        for(int i = 0; i < 10; ++i){
            int r = 0;
            int g = 0;
            int b = 0;
            for(int j = 0; j < m; ++j){
                if(rings[2*j+1] - '0' == i){
                    if(rings[2*j] == 'R'){
                        r = 1;
                    }
                    else if(rings[2*j] == 'G'){
                        g = 1;
                    }
                    else{
                        b = 1;
                    }
                }
            }
            ans += r + g + b == 3;
        }
        return ans;
    }
};

File: Rust/Easy/Accepted/rings-and-rods_8.rs
impl Solution {
    pub fn count_points(rings: String) -> i32 {
        
        let mut result = 0;
        let mut rods = [[false; 3]; 10];
        for i in 0..rings.len() / 2 {
            rods[rings.chars().nth(i * 2 + 1).unwrap() as usize - 48][match rings.chars().nth(i * 2).unwrap() {
                'R' => 0,
                'G' => 1,
                'B' => 2,
                _ => 0,
            }] = true;
        }
        for i in 0..10 {
            if rods[i][0] && rods[i][1] && rods[i][2] {
                result += 1;
            }
        }
        result
    }
}